### ISSUE TITLE
Revise .cicdtemplate for Github Actions

### ISSUE THREAD

**Issue by hoangnguyen92dn** [nimblehq/android-templates#539]
## Why

There are some points to revamp the current CI/CD template to have workflow ready to work in generated project: 
https://github.com/nimblehq/android-templates/tree/develop/.cicdtemplate/.github

- [ ] Remove unused templates since it's out of date and unmantainced, the team need to reference to [git-template](https://github.com/nimblehq/git-template/tree/main/.github) as single source truth and latest update for new projects: 

![image](https://github.com/nimblehq/android-templates/assets/6950766/870f8b5f-a009-4408-b3fa-ba1a303b64fa)

-----

- [ ] Combine reviewing PR workflows: `review_pull_request.yml` && `run_detekt_and_unit_tests.yml`. Both workflow have the same steps can be extract to shared workflow to reuse (Reference: https://docs.github.com/en/actions/using-workflows/sharing-workflows-secrets-and-runners-with-your-organization).

![image](https://github.com/nimblehq/android-templates/assets/6950766/ce8e59e5-c06e-42a5-b01e-9fa8399993fa)

-----

- [ ] Revamp the deployment workflow: `deploy_staging_and_production_to_firebase_app_distribution.yml`. Github supports environment for deployment therefore creating variables/secretes with environment suffix is unnecessary (Reference: https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment)

![image](https://github.com/nimblehq/android-templates/assets/6950766/4bd16671-6f5b-4493-8b84-eb39cb968c77)

## Who Benefits?

Developers

**Comment by luongvo** [comment]
> Remove unused templates since it's out of date and unmantainced, the team need to reference to [git-template](https://github.com/nimblehq/git-template/tree/main/.github?rgh-link-date=2023-10-09T02%3A58%3A50Z) as single source truth and latest update for new projects:

@hoangnguyen92dn Instead of `remove`, IMO, we should _keep them up to date_ or _have a script to sync them when generating new projects_.

> Revamp the deployment workflow: deploy_staging_and_production_to_firebase_app_distribution.yml. Github supports environment for deployment therefore creating variables/secretes with environment suffix is unnecessary

Agree, we should separate Environment variables by env: production and staging instead of by template: template-compose and template-xml.

**Comment by hoangnguyen92dn** [comment]
@luongvo In fact, when the team create new repo, I see that choosing `git-template` from the options would be more efficient 😅

**Comment by luongvo** [comment]
@hoangnguyen92dn Yes, but then, we will generate and push up the generated initial project from our Template.
